site/cypress/e2e/pdfs.spec.js

```diff
@@ -0,0 +1,105 @@
+import {getCurrentSemester} from '../support/utils.js';
+
+function switch_settings(setting, pdf_type) {
+    cy.visit(`/courses/${getCurrentSemester()}/sample`);
+    cy.login('instructor');
+    cy.get(`a[href*="/sample/gradeable/${pdf_type}/update"]`).click();
```

Comment:
You don't need to click on links like this.  Just navigate to the page you want to cut down on the overall running time.  Apply throughout.